src/lib/types.ts

```diff
@@ -209,14 +210,319 @@ type ResolveInviteCodeResponse = {
+};
+
+type SpaceFields = {
+  type: "physical" | "gps" | "virtual";
+  name: string;
+  location: PhysicalLocation | GPSLocation | VirtualLocation;
+  capacity: number;
+  accessibility: string;
+  description: string;
+  contact: string;
+  message: string;
+  link: Link;
+  images: Image[];
+  availability: TimeSpan[] | "always";
+  multiBookable: boolean;
+};
+
+type ResourceFields = {
+  name: string;
+  ownerId: PublicKey;
+  description: string;
+  contact: string;
+  link: Link;
+  images: Image[];
+  availability: TimeSpan[] | "always";
+  multiBookable: boolean; // resource can be booked more than once in the same time span
+};
+
+type EventFields = {
+  name: string;
+  description: string;
+  location?: SpaceRequestId; // ref to a space
+  startDate: Date; // allocated time of a space
+  endDate: Date; // allocated time of a space
+  publicStartDate?: Date; // public facing
+  publicEndDate?: Date; // public facing
+  resources: ResourceRequestId[];
+  links: Link[];
+  images: Image[];
+};
+
+/**
+ * User
+ *
+ * NOTE: sent on `inbox` channel
+ */
+
+type UserNameAssigned = {
+  type: "user_name_assigned";
   data: {
     name: string;
-    startDate?: string;
-    endDate?: string;
+  };
+};
+
+/**
+ * Access
+ *
+ * NOTE: sent on `inbox` channel
+ */
+
+type CalendarAccessRequested = {
+  type: "calendar_access_requested";
+  data: {
+    calendarId: Hash;
+    name: string;
+    message: string;
+  };
+};
+
+type CalendarAccessAccepted = {
+  type: "calendar_access_accepted";
+  data: {
+    calendarId: Hash;
+  };
+};
+
+type CalendarAccessRejected = {
+  type: "calendar_access_rejected";
+  data: {
+    calendarId: Hash;
+  };
+};
+
+/**
+ * Calendar
+ */
+
+type CalendarCreated = {
+  type: "calendar_created";
+  data: {
+    fields: CalendarFields;
+  };
+};
+
+type CalendarUpdated = {
+  type: "calendar_updated";
+  data: {
+    id: Hash;
+    fields: CalendarFields;
+  };
+};
+
+type SpacesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type ResourcesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type AboutPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type CalendarDeleted = {
+  type: "calendar_deleted";
+};
+
+/**
+ * Space
+ */
+
+type SpaceCreated = {
+  type: "space_created";
+  data: {
+    field: SpaceFields;
+  };
+};
+
+type SpaceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: SpaceFields;
+  };
+};
+
+type SpaceDeleted = {
+  type: "space_deleted";
+};
+
+/**
+ * Resource
+ */
+
+type ResourceCreated = {
+  type: "resource_created";
+  data: {
+    fields: ResourceFields;
+  };
+};
+
+type ResourceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: ResourceFields;
+  };
+};
+
+type ResourceDeleted = {
+  type: "resource_deleted";
+};
```

Comment:
This needs an `id` as well, right?

Reply:
Well, we weren't consistent with this anyway though, as the id is present in other update messages :smile: but in any case it's an interesting point that this id is repeated in the protocol message and p2panda header extension which "wraps" it.